src/net/README.md

```diff
@@ -83,76 +75,71 @@ and generate the `next` generation.  However, as mentioned previously,
 when moving leaves between parents we must do so in the dependency order
 of the current generation.
 
-So, the user (sysrepo) needs to add `.dn` scripts in the current tree,
-and `.up` scripts in the next.
+So, the user (sysrepo) needs to add `cmd.exit` scripts in the current
+tree for every leaf that leaves a parent, and `cmd.init` scripts for
+leaves that are new or added to parents in the next generation.
 
 In our example, interface `eth4` is moved from `lag0` to `br0`, so we
-need to run `eth4/ip-link.dn` in the current generation first to remove
-`eth4` from `lag0` before its `ip-link.up` script in the next generation
+need to run `eth4/ip.exit` in the current generation first to remove
+`eth4` from `lag0` before its `ip.init` script in the next generation
 sets `eth4` as a bridge member instead.
 
-We traverse the current generation and execute all `.dn` scripts:
+We traverse the current generation and execute all `cmd.exit` scripts:
 
     /run/net/<GEN>/
-	   |-- lo/
-	   |    `-- deps/
 	   |-- br0/
 	   |    |-- deps/
 	   |    |    |-- eth1 -> ../../eth1
 	   |    |    |-- eth2 -> ../../eth2
 	   |    |    |-- eth3 -> ../../eth3
 	   |    |    `-- lag0 -> ../../lag0
-	   |    `-- ip-link.up
+	   |    `-- ip.init
 	   |-- eth0/
 	   |    |-- deps/
-	   |    |-- ip-link.up
-	   |    `-- ip-addr.up
+	   |    `-- ip.init
 	   |-- eth4/
 	   |    |-- deps/
-	   |    |-- ip-link.dn
-	   |    `-- ip-link.up
+	   |    |-- ip.exit
+	   |    `-- ip.init
 	   |-- lag0/
 	   |    |-- deps/
 	   |    |    |-- eth4 -> ../../eth4
 	   |    |    `-- eth5 -> ../../eth5
-	   |    `-- ip-link.up
+	   |    `-- ip.init
 	   `-- vlan1/
 	        |-- deps/
 	        |    `-- br0 -> ../../br0
-	        |-- ip-link.up
-	        `-- ip-addr.up
+	        `-- ip.init
 
-Now we can run all the `.up` scripts in the next generation:
+Now we can run all the `cmd.init` scripts in the next generation:
 
     /run/net/<GEN+1>/
-	   |-- lo/
-	   |    |-- deps/
-	   |    |-- ip-link.up
-	   |    `-- ip-addr.up
 	   |-- br0/
 	   |    |-- deps/
 	   |    |    |-- eth1 -> ../../eth1
 	   |    |    |-- eth2 -> ../../eth2
 	   |    |    |-- eth3 -> ../../eth3
 	   |    |    |-- eth4 -> ../../eth4
 	   |    |    `-- lag0 -> ../../lag0
-	   |    `-- ip-link.up
+	   |    `-- ip.init
 	   |-- eth0/
 	   |    |-- deps/
-	   |    |-- ip-link.up
-	   |    `-- ip-addr.up
+	   |    `-- ip.init
 	   |-- eth4/
 	   |    |-- deps/
-	   |    |-- ip-link.dn
-	   |    `-- ip-link.up
+	   |    `-- ip.init
 	   |-- lag0/
 	   |    |-- deps/
 	   |    |    `-- eth5 -> ../../eth5
-	   |    `-- ip-link.up
+	   |    `-- ip.init
 	   `-- vlan1/
 	        |-- deps/
 	        |    `-- br0 -> ../../br0
-	        |-- ip-link.up
-	        `-- ip-addr.up
-
-
+	        `-- ip.init
+
+When there are no changes compared to the previous generation, the
+`cmd.init` scripts can be empty.  The existence of an `ip.init` script,
+however, means that it is allowed to be brought up on `net up`.  The
```

Comment:
Not sure we can infer that. In a single config generation, a previously "admin up" interface could be set to "admin down" _and_ be added to a bridge. In such a scenario, there will be an `ip.init`, but we are still not allowed to bring it up on `net up`.

Reply:
OK, makes sense.  I'll go with the admin-state suggestion below and add test for it.